### PR TITLE
Support :prefix on index rename

### PR DESCRIPTION
**Pull request by maltoe**
Hi :wave: 

we wanted to rename an index in a schema on Postgres like the following but the prefix was ignored. This patch adds support for `:prefix` for Postgres.

```elixir
rename index(:foo, [:column], prefix: "bar", name: "old"), to: "new"
```

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
A separate PR is welcome, please!

**Comment by josevalim** [comment]
We would have to update the lock file, but we did that on main anyway as part of v3.11.0. :)

**Comment by maltoe** [comment]
> We would have to update the lock file, but we did that on main anyway as part of v3.11.0. :)

Right, the lock file is in the cache key, didn't see that. 

Not sure I get the rest of your statement though :see_no_evil: - is there anything else I can do for this PR?

**Comment by josevalim** [comment]
This PR is good, I am just waiting for a second opinion (who can go ahead and merge it!).

File: lib/ecto/adapters/postgres/connection.ex
Comment (greg-rychlewski) [inline lib/ecto/adapters/postgres/connection.ex:1261]: Maybe we should combine `quote_table` and `quote_name`?  Using `quote_table` here will produce a confusing error

```elixir
defp quote_table(name) do
  if String.contains?(name, "\"") do
    error!(nil, "bad table name #{inspect(name)}")
  end

  [?", name, ?"]
end
```

Comment (maltoe) [reply lib/ecto/adapters/postgres/connection.ex:1261]: It's also used like this in the create/drop index functions, but :+1: 

As a suggestion, how about dropping `quote_table/1` entirely and using `quote_name/1` in `quote_table/2`? As the message in `quote_name/1` also includes tables, and using `quote_table/1` for the prefix as currently done in `quote_table/2` doesn't really make sense either.

Should this be done in this PR?

Comment (greg-rychlewski) [reply lib/ecto/adapters/postgres/connection.ex:1261]: To me it looks like the only difference between `quote_table` and `quote_name` is the error message and the former can use a prefix. So I think getting rid of `quote_table` altogether probably makes sense? Unless @josevalim  is seeing a reason to keep it.

**Comment by greg-rychlewski** [comment]
Jose I think you may have forgotten to push the 3.11 commit to github for Ecto SQL

File: lib/ecto/adapters/postgres/connection.ex
Comment (josevalim) [reply lib/ecto/adapters/postgres/connection.ex:1261]: As you prefer @greg-rychlewski!

**Comment by josevalim** [comment]
> Jose I think you may have forgotten to push the 3.11 commit to github for Ecto SQL

Done!

File: lib/ecto/adapters/postgres/connection.ex
Comment (greg-rychlewski) [reply lib/ecto/adapters/postgres/connection.ex:1261]: @maltoe Sorry to add more to the PR, would you be able to get rid of all the quote_table functions and use this instead:

```elixir
defp quote_name(nil, name), do: quote_name(name)

defp quote_name(prefix, name), do: [quote_name(prefix), ?., quote_name(name)]

defp quote_name(name) when is_atom(name) do
  quote_name(Atom.to_string(name))
end

defp quote_name(name) when is_binary(name) do
  if String.contains?(name, "\"") do
    error!(nil, "bad literal/field/index/table name #{inspect(name)} (\" is not permitted)")
  end

  [?", name, ?"]
end
```

Then we can merge =)

Comment (maltoe) [reply lib/ecto/adapters/postgres/connection.ex]: @greg-rychlewski Done & rebased on `master`

**Review by greg-rychlewski — APPROVED** [review]

**Comment by greg-rychlewski** [comment]
Thanks!

**Pull request merged**
